rtt/base/InputPortInterface.cpp

```diff
@@ -84,6 +84,11 @@ InputPortInterface::NewDataOnPortEvent* InputPortInterface::getNewDataOnPortEven
 #endif
 bool InputPortInterface::connectTo(PortInterface* other, ConnPolicy const& policy)
 {
+    if(!other)
+    {
+        log(Error) << "Function 'connectTo()' could not connect InputPort " << getName() << " to a non existent port." << endlog();
+        return false;
+    }
```

Comment:
The check after the dynamic_cast below should be sufficient. Null pointers are always casted to null pointers.